spec/index.emu

```diff
@@ -74,6 +74,7 @@ location: https://rawgit.com/tc39/proposal-realms/master/index.html
         1. Set _O_.[[Realm]] to _realmRec_.
         1. Perform ? SetRealmGlobalObject(_realmRec_, *undefined*, *undefined*).
         1. Perform ? SetDefaultGlobalBindings(_O_.[[Realm]]).
+        1. Create any implementation-defined global object properties on _globalObj_.
```

Comment:
matching https://tc39.es/ecma262/#sec-initializehostdefinedrealm - step 11

Reply:
maybe the common steps between this, and InitializeHostDefinedRealm, could be factored out into a shared abstract operation?

Reply:
yes, we discussed this extensibly today in a bigger context. @erights raised valid concerns about clause 18 (https://tc39.es/ecma262/#sec-global-object) last bullet point:

> may have host defined properties in addition to the properties defined in this specification. This may include a property whose value is the global object itself.

That is part of `SetDefaultGlobalBindings` abstract operation (which iterates over every property of each property of the Global Object as defined by clause 18.

The two main concerns are:

1. "implementation-defined properties" vs "host defined properties". that on itself is confusing.
2. having that as part of the global object definition makes it hard for compartments, and other ways to create a global object in the future if we want to introduce user-defined properties via some hooks.

The recommendation at this point is to create a new abstract operation, and call that right after `SetDefaultGlobalBindings()`, and remove the last bullet of clause 18 entirely. This change has no impact on implementation, it is just spec refactor.

@ljharb @littledan can we do that on a separate PR for 262? or should we continue with that here?

Reply:
imo anything that can hit 262 separate from a proposal, should.

Reply:
@ljharb what? Can you rephrase?

Reply:
:Any editorial refactorings that can land in 262 without consensus, and make nonzero sense on their own, should imo land prior to the proposal".

Reply:
good, thanks.

Reply:
Ok, I can send the PR for 262, and we can discussed there. I will keep this PR around until we get to consensus.